src/context/data/videoContext.js

```diff
@@ -0,0 +1,58 @@
+import axios from "axios";
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { ACTION_TYPE } from "../../utils/constant";
+import { initialState, videoReducer } from "../../reducer/videoReducer";
+
+const DataContext = createContext();
+
+const DataProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(videoReducer, initialState);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const {
+          data: { categories },
+        } = await axios.get("/api/categories");
```

Comment:
Here if your api call for category will fail, then videos api will also not work. You can use separate effect hooks for individual apis.

Reply:
Okay , will refactor it